internal/services/instance_types_service.go

```diff
@@ -23,6 +24,10 @@ func ListInstanceTypes(w http.ResponseWriter, r *http.Request) {
 
 	arn, err := fetchARN(r.Context(), sourcesClient, sourceId)
 	if err != nil {
+		if errors.Is(err, sources.ApplicationNotFoundErr) || errors.Is(err, sources.AuthenticationForSourcesNotFoundErr) {
+			renderError(w, r, payloads.SourcesClientError(r.Context(), "can't fetch arn from sources", err, 404))
+			return
+		}
 		renderError(w, r, payloads.New3rdPartyClientError(r.Context(), "can't fetch arn from sources", err))
```

Comment:
I actually suggest to get rid of New3rdPartyClientError alltogether and only keep SourcesClientError with 404 or 500.

Reply:
What about other clients like ec2/sts?

Reply:
Yeah I thought to have separate errors per each service.